Samples/NetworkMediaStreamer/CameraRTPStreamerApp/CameraRTPStreamerApp.cpp

```diff
@@ -0,0 +1,430 @@
+{
+    STARTUPINFO si, si1;
+    PROCESS_INFORMATION pi;
+    GetStartupInfo(&si1);
+    ZeroMemory(&si, sizeof(si));
+    si.cb = sizeof(si);
+    ZeroMemory(&pi, sizeof(pi));
+    std::wstring cmdline = L"powershell Get-Content " + filename + L" -Wait ";
+    std::wstring title = L"RTSPServer Log Console";
+    si.lpTitle = (LPWSTR)title.c_str();
+
+    si.dwX = si1.dwX + 500;
+    si.dwY = si1.dwY;
+    si.dwXCountChars = 40;
+    si.dwYCountChars = 10;
+    si.dwFlags = STARTF_USESHOWWINDOW | STARTF_USEPOSITION | STARTF_PREVENTPINNING | STARTF_USECOUNTCHARS;
+    si.wShowWindow = SW_SHOWNOACTIVATE;
+    // Start the child process. 
+    if (!CreateProcess(NULL,   // No module name (use command line)
+        (LPWSTR)cmdline.c_str(),        // Command line
+        NULL,           // Process handle not inheritable
+        NULL,           // Thread handle not inheritable
+        FALSE,          // Set handle inheritance to FALSE
+        CREATE_NEW_CONSOLE,              // No creation flags
+        NULL,           // Use parent's environment block
+        NULL,           // Use parent's starting directory 
+        &si,            // Pointer to STARTUPINFO structure
+        &pi)            // Pointer to PROCESS_INFORMATION structure
+        )
+    {
+        check_win32(GetLastError());
+    }
+    return pi;
+
+}
+
+void StopLoggerConsole(PROCESS_INFORMATION pi)
+{
+    TerminateProcess(pi.hProcess, 0);
+    // Close process and thread handles. 
+    CloseHandle(pi.hProcess);
+    CloseHandle(pi.hThread);
+}
+
+winrt::hstring GetDeviceSelectionFromUser()
+{
+    auto filteredDevices = DeviceInformation::FindAllAsync(DeviceClass::VideoCapture).get();
+    if (!filteredDevices.Size())
+    {
+        throw_hresult(MF_E_NO_CAPTURE_DEVICES_AVAILABLE);
+    }
+    int selection = 1;
+    auto deviceIter = filteredDevices.First();
+    while (deviceIter.HasCurrent())
+    {
+        auto device = deviceIter.Current();
+        std::wcout << std::endl << selection++ << L". " << device.Name().c_str() << L":" << device.Id().c_str();
+        deviceIter.MoveNext();
+    }
+    std::cout << "\nEnter Selection:";
+    std::cin >> selection; selection--;
+    return filteredDevices.GetAt(selection).Id();
+}
+
+void GetUserSelectionAndSetMediaFormat(MediaCapture& mc)
+{
+    auto formats = mc.VideoDeviceController().GetAvailableMediaStreamProperties(MediaStreamType::VideoRecord);
+    IMediaEncodingProperties selectedProp(nullptr);
+    uint32_t idx = 1;
+    for (auto f : formats)
+    {
+        auto format = f.try_as<VideoEncodingProperties>();
+        float fr = (float)format.FrameRate().Numerator() / (float)format.FrameRate().Denominator();
+        std::wcout << L"\n" << idx++ << L". " << format.Width() << L"x" << format.Height() << L"@" << /*std::setprecision(2) <<*/ fr << L":" << format.Subtype().c_str();
+    }
+    std::cout << "Enter format choice: ";
+    std::cin >> idx; idx--;

```

Comment:
should there be a bound check on the user input value to ensure it's within the list?

Reply:
i can add a loop blocking the user to input within bounds. However, this is just a test application, and the user here is a tester and its ok to throw out-of-bounds exception and exit the program, which is done by the subsequent API calls using the user choice.   I can add an explicit check here if it makes it more readable.

---
In reply to: [493959390](https://github.com/microsoft/Windows-Camera/pull/13#discussion_r493959390) [](ancestors = 493959390)